dbmigrator/utils.py

```diff
@@ -206,6 +210,18 @@ def run_migration(cursor, version, migration_name, migration):
     except AttributeError:
         # not a repeat migration
         pass
+
+    if not run_defers:
+        try:
+            if migration.up.dbmigrator_deferrable:
+                print('Marking migration {} {} as deferred'
+                      .format(version, migration_name))
+                mark_migration(cursor, version, 'deferred')
```

Comment:
I was curious, what is the advantage of marking them in the DB before they run? 

I would have expected that `--run-deferred` would run `should_run()` on every `@deferred` migration and then run `up()`, no?